### PR TITLE
Update test coverage for ODataCollectionReader.

### PR DESCRIPTION
**Pull request by biaol-odata**
### Issues
*Update test coverage for ODataCollectionReader.*  

### Description
*Improve coverage by adding test case and remove dead code.*

### Checklist (Uncheck if it is not completed)
- [ x ] Test cases added
- [ x ] Build and test with one-click build and test script passed

### Additional work necessary
*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: test/FunctionalTests/Tests/DataOData/Tests/OData.Reader.Tests/MessageToObjectModelReader.cs [hunk 1]
```diff
@@ -615,7 +627,7 @@ private ODataBatchOperation GetRequestOperation(ODataBatchReaderTestWrapper batc
                 {
                     expectedPartElement = odataRequest.Body == null ? null : odataRequest.Body.RootElement;
                 }
-                
+
                 ODataBatchOperationRequestMessage requestMessage = batchReader.CreateOperationRequestMessage();
```
Comment (robward-ms) [inline test/FunctionalTests/Tests/DataOData/Tests/OData.Reader.Tests/MessageToObjectModelReader.cs]: Nit: should be 1 blank line.

**Comment by robward-ms** [comment]
:shipit:

File: test/FunctionalTests/Tests/DataOData/Tests/OData.Reader.Tests/ReaderTestConfiguration.cs [hunk 2]
```diff
@@ -70,18 +74,24 @@ public ReaderTestConfiguration(ReaderTestConfiguration other)
         public bool Synchronous { get; private set; }
 
         /// <summary>
+        /// True if test to skip reader state validation beofer reading.
```
Comment (AlanWong-MS) [inline test/FunctionalTests/Tests/DataOData/Tests/OData.Reader.Tests/ReaderTestConfiguration.cs]: Typo: beofer -> before

File: test/FunctionalTests/Tests/DataOData/Tests/OData.Reader.Tests/MessageToObjectModelReader.cs (hunk 1 above)
Comment (biaol-odata) [inline test/FunctionalTests/Tests/DataOData/Tests/OData.Reader.Tests/MessageToObjectModelReader.cs]: # of blank line is still 1. This is just to remove the tailing whitespaces of the blank line.

File: test/FunctionalTests/Tests/DataOData/Tests/OData.Reader.Tests/ReaderTestConfiguration.cs (hunk 2 above)
Comment (biaol-odata) [inline test/FunctionalTests/Tests/DataOData/Tests/OData.Reader.Tests/ReaderTestConfiguration.cs]: fixed.

**Pull request merged**